### DISCUSSION TITLE
CLI KaTeX-type tool to render an entire HTML page

### DISCUSSION THREAD

**Discussion by pgeipi10** [KaTeX/KaTeX#3239] [Q&A]
Is there a tool, such as CLI KaTeX, that could render an entire HTML page, e.g. katex -i  chapter1.html

**Comment by edemaine** [comment]
I don't think such a tool exists, but it sounds like a nice thing to build. It shouldn't be *too* hard to add a feature/option to `cli.js`, or build a new tool, that uses something like [renderMathInText](https://github.com/KaTeX/KaTeX/blob/master/contrib/auto-render/auto-render.js) to render all math expressions (though one would have to specify the desired delimiters too).  Hmm, renderMathInText requires a DOM, so this could be done via `xmldom` or similar.  Or perhaps simpler would be to finally implement #604 and use that for text-based delimiter finding.
